File: guides/replace-h2-with-real-database-for-testing/jdbc/java/src/main/java/example/micronaut/Product.java
package example.micronaut;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity("products") // <1>
public class Product {
    @Id // <2>
    private Long id;

    private String code;

    private String name;

    public Product(Long id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: guides/replace-h2-with-real-database-for-testing/jdbc/python/src/example/micronaut/product.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import Id, MappedEntity


@dataclass
@MappedEntity("products")  # <1>
class Product:
    id: Annotated[int, Id]  # <2>
    code: str
    name: str
